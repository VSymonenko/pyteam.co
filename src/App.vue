<template lang="pug">
  transition(name="app")
    v-app(:style={overflow: 'hidden'} v-show="show")
      AppBar(:title="appBar.title" :clipped="appBar.clipped" :menu="appBar.tabs")
      v-content
        AwesomeSwiper(:tab-routes="appBar.tabs" :colors="colors")
          router-view
      v-footer(:fixed="fixed" app)
        span(:style="{ padding: '10px'}") &copy; 2018
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';
import AppBar from './components/AppBar.vue';
import { mapGetters } from 'vuex';
import AwesomeSwiper from '@/components/AwesomeSwiper.vue';

@Component({
  name: 'App',
  components: {
    AppBar,
    AwesomeSwiper,
  },
  computed: {
    ...mapGetters([
      'appBar',
      'colors',
    ]),
  },
})

export default class App extends Vue {
  public fixed: boolean = false;
  private darkTheme: boolean = false;
  private show: boolean = false;

  private mounted(): void {
    setTimeout(() => this.show = true, 500);
  }
}
</script>

<style lang="stylus">
::-webkit-scrollbar
  display: none
.app-enter-active, .app-leave-active {
  transition: opacity .5s;
}
.app-enter, .app-leave-to /* .app-leave-active  */ {
  opacity: 0;
}
</style>
