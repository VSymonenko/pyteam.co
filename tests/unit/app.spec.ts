import App from '@/App.vue';
import { shallowMount } from '@vue/test-utils';
import './common';

test('renders correctly', () => {
  const wrapper = shallowMount(App);

  expect(wrapper.isVueInstance()).toBeTruthy();
  expect(wrapper).toMatchSnapshot();
});
