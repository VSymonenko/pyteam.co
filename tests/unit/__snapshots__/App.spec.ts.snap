// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<vapp-stub id="app" style="overflow: hidden; display: none;" name="app">
  <appbar-stub title="PyTeam" menu="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"></appbar-stub>
  <vcontent-stub tag="main">
    <awesomeswiper-stub tabroutes="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" colors="hsla(270, 50%, 40%, 1),hsla(35, 100%, 67%, 1),hsla(48, 86%, 72%, 1),hsla(11, 100%, 87%, 1),hsla(176, 100%, 44%, 1),hsla(171, 44%, 69%, 1)">
      <router-view-stub name="default"></router-view-stub>
    </awesomeswiper-stub>
  </vcontent-stub>
  <vfooter-stub height="32" app="app"><span style="padding: 10px;">Â© 2018</span></vfooter-stub>
</vapp-stub>
`;
